### PR TITLE
Fix for Issue#252

### PR DESCRIPTION
**Pull request by kpanwar**
Fix for Issue#252
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/253%23discussion_r24055732%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/253%23discussion_r24625525%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/253%23issuecomment-74170777%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/253%23issuecomment-74170777%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Signed-off%22%2C%20%22created_at%22%3A%20%222015-02-12T22%3A50%3A21Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/10979061%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ranyelhousieny%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20ffe17b2ce8b8965160ecce39e9750fcc35c06c5c%20ADALiOS/ADALiOS/ADClientMetrics.m%2028%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/253%23discussion_r24055732%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22It%27s%20still%20not%20a%20bad%20idea%20to%20nil%20check%20these%20things%20just-in-case.%22%2C%20%22created_at%22%3A%20%222015-02-04T01%3A11%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle-MS%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-02-12T22%3A10%3A15Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5713915%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kpanwar%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20ADALiOS/ADALiOS/ADClientMetrics.m%3AL84-102%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/253#issuecomment-74170777'>General Comment</a></b>
- <a href='https://github.com/ranyelhousieny'></a> Signed-off
- [x] <a href='#crh-comment-Pull ffe17b2ce8b8965160ecce39e9750fcc35c06c5c ADALiOS/ADALiOS/ADClientMetrics.m 28'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/253#discussion_r24055732'>File: ADALiOS/ADALiOS/ADClientMetrics.m:L84-102</a></b>
- <a href='https://github.com/RPangrle-MS'></a> It's still not a bad idea to nil check these things just-in-case.

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-objc/pull/253?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-objc/pull/253?approve=1'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/253'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by brandwe** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/253'>Refresh</a>.<a href='#crh-update-none'></a>

File: ADALiOS/ADALiOS/ADClientMetrics.m [hunk 1]
```diff
@@ -88,12 +84,18 @@ - (void) beginClientMetricsRecordForEndpoint: (NSString*) endPoint
             return;
         }
         if(_isPending){
-        [requestHeader setObject:_errorToReport forKey:HeaderLastError];
-        [requestHeader setObject:_responseTime forKey:HeaderLastResponseTime];
-        [requestHeader setObject:[ADHelpers getEndpointName:_endpoint] forKey:HeaderLastEndpoint];
-        [requestHeader setObject:_correlationId forKey:HeaderLastRequest];
+            [requestHeader setObject:_errorToReport forKey:HeaderLastError];
```
Comment (RPangrle-MS) [inline ADALiOS/ADALiOS/ADClientMetrics.m]: It's still not a bad idea to nil check these things just-in-case.

Comment (kpanwar) [inline ADALiOS/ADALiOS/ADClientMetrics.m]: :+1:

**Comment by ranyelhousieny** [comment]
Signed-off

**Comment by ranyelhousieny** [comment]
Signed-off

**Pull request merged**